outlines/text/parsing.py

```diff
@@ -383,24 +385,30 @@ def map_partial_states_to_vocab(
     """
 
     final_state_string_idx = None
+
     # Partial parse states to the subsets of the vocabulary that accept them
     pstate_to_vocab = defaultdict(set)
+    possible_paths = {}
     for symbol_name, fsm in terminals_to_fsms_map.items():
+        terminal_possible_paths = defaultdict(set)
         for i, vocab_string in enumerate(vocabulary):
             if vocab_string == final_state_string:
                 final_state_string_idx = i
 
             for end_idx, state_seq in find_partial_matches(fsm, vocab_string):
                 if partial_match_filter(vocab_string, end_idx, state_seq):
+                    terminal_possible_paths[state_seq[0]].add(state_seq[-1])
```

Comment:
Whether or not the last state is a final/accept state of the FSM will depend on `partial_match_filter`, no?  We might need extra handling for this, or we could track these things in that filter function, after adding the symbol name as an argument, of course.

Reply:
The check would fail if the combination of the vocabulary and `partial_match_filter` didn't allow to reach a terminal state, no? Isn't that the desired behaviour?